examples/threading/threading.c

```diff
@@ -28,6 +67,33 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      * 
      * See implementation details in threading.h file comment block
      */
+    int ret;	
+	struct thread_data *pThreadParam;
+	
+	//Allocate memeory for data structure
+	pThreadParam = (struct thread_data *)malloc(sizeof(struct thread_data));
+	if(pThreadParam == NULL){
+		ERROR_LOG("Not enough memory to be allocated for thread_data");
+		return false;
+	}
+	
+	//Setup structure parameters
+	pThreadParam->pmutex = mutex;
+	pThreadParam->wait_ms_before_obtain = wait_to_obtain_ms;
+	pThreadParam->wait_ms_before_release = wait_to_release_ms;
+
+	//Create thread
+	//int pthread_create (pthread_t *thread,
+	//const pthread_attr_t *attr,
+	//void *(*start_routine) (void *),
+	//void *arg);
+	ret = pthread_create(thread, NULL, threadfunc, (void *)pThreadParam);
+	if(ret == 0){
+		DEBUG_LOG("Thread created successfully");
+		return true;
+	}
+	
+	ERROR_LOG("FAILED while creating the thread");     
     return false;
```

Comment:
@ruchitnaik who frees `pThreadParam` in this code path?

Reply:
It should be freed here before returning the false. Thanks for pointing it out.